src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1609,43 +1612,18 @@ static bool addBsonFilter
   }
   else if (opr == OPR_EXIST)
   {
-    if (std::string(right) == ENT_ENTITY_TYPE)
-    {
-      // Special case: entity type
-      k = std::string("_id.") + ENT_ENTITY_TYPE;
+    k = std::string(ENT_ATTRS) + "." + right;
 
-      bb.append("$exists", true).append("$ne", "");
-      bob.append(k, bb.obj());
-      f = bob.obj();
-    }
-    else
-    {
-      // Regular attribute
-      k = std::string(ENT_ATTRS) + "." + right;
-
-      bb.append("$exists", true);
-      bob.append(k, bb.obj());
-      f = bob.obj();
-    }
+    bb.append("$exists", true);
+    bob.append(k, bb.obj());
+    f = bob.obj();
   }
   else if (opr == OPR_NOT_EXIST)
   {
-    if (std::string(right) == ENT_ENTITY_TYPE)
-    {
-      // Special case: entity type
```

Comment:
Not really part of this PR, but a bit of cleanup regarding `+type` removal.

NTC